core/store/src/db/testdb.rs

```diff
@@ -19,6 +18,18 @@ pub struct TestDB {
     // The TestDB doesn't produce any stats on its own, it's up to the user of
     // this class to set the stats as they need it.
     stats: RwLock<Option<StoreStatistics>>,
+
+    cache: Arc<deserialized_column::Cache>,
```

Comment:
Is it a strict requirement for every db to have this cache? Some like the future cloud db (write only for most purposes) may be better off without it. 

It's fine here, it's good to have the cache for tests.

Reply:
Not a requirement, e.g. `ColdDB` does not implement `Database::deserialized_cache` so it doesn't really need to keep track of anything either.